quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java

```diff
@@ -98,6 +107,39 @@ public RawKeycloakDistribution(boolean debug, boolean manualStop, boolean enable
         this.reCreate = reCreate;
         this.removeBuildOptionsAfterBuild = removeBuildOptionsAfterBuild;
         this.distPath = prepareDistribution();
+
+        String hostsFile = System.getProperty("jdk.net.hosts.file");
+        if (hostsFile != null) {
+            // Check that our hostname is present in the 'hosts' file for the test to avoid a lot of warnings in the logs
+            // which might then hide real problems in the test.
+            try {
+                //noinspection ResultOfMethodCallIgnored
+                InetAddress.getLocalHost();
+            } catch (UnknownHostException ex) {
+                //
+                Pattern pattern = Pattern.compile("Unable to resolve host (?<host>[^ ]*) in hosts");
+                String message = ex.getMessage();
+                Matcher matcher = pattern.matcher(message);
+                if (matcher.find()) {
+                    try (BufferedWriter bw = new BufferedWriter(new FileWriter(hostsFile, true))) {
+                        bw.newLine();
+                        bw.write("127.0.0.1 " + matcher.group("host"));
+                    } catch (IOException e) {
+                        throw new RuntimeException("unable to append to file " + hostsFile, e);
+                    }
+                } else {
+                    throw new RuntimeException("unable to parse host name from message '" + message + "'");
+                }
+
+                // now the resolving of the hostname should work as expected
+                try {
+                    //noinspection ResultOfMethodCallIgnored
+                    InetAddress.getLocalHost();
+                } catch (UnknownHostException e) {
+                    throw new RuntimeException("unable to resolve host name even after adding it to the host_name file", e);
+                }
+            }
+        }
```

Comment:
@ahus1 I'm not sure if we can override the 'jdk.net.hosts.file' hosts file without any fallback. For GHA CI it's probably acceptable, but not sure if we can do it when a user execute these tests on a local machine/server. 

Moreover, can you point me to some job, where the `Unable to resolve host` warnings are printed? AFAIK, for the FipsDistTest, there are not such msgs and I haven't spotted it for a long time. AFAIK, these warnings should be already gone after merging this PR: https://github.com/keycloak/keycloak/pull/22035.

Nitpick: Potentially, I'd separate it into a new self-descriptive method as it might be a little bit confusing to have written such a functionality in the constructor, right?

Reply:
You're right, this was already fixed. I think I was mislead by the login the original issue. 

I saw in this run that those exceptions are no longer there: https://github.com/keycloak/keycloak/actions/runs/7139549047/job/19451789357

I removed that part, and only kept the bit that shows the original exception.